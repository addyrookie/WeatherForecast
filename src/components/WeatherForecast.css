* {
  margin: 0;
  border: 0;
  box-sizing: border-box;
}

body {
	  background-color: teal;
}



.heading {
	padding: 20px;
}

.weather-data-container {
  background-color: teal;
  padding: 5px;
  margin: 0 ;
}

.weather-data-container li {
  list-style: none;
}

.weather-item-list li{
	border: 1px solid #FFD700;
	padding: 0px;
}


@media all and (min-width: 200px) {

	.weather-item-list {
		display: grid;
		grid-template-columns: repeat(6, minmax(70px,1fr));
	}
}