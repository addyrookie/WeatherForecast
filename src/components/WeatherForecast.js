import React from "react";
import _ from "lodash";
import './WeatherForecast.css';

export class WeatherForecast extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			weatherData: {},
			city: ''
		}
	}

	onSubmitCity = (event) =>  {
		event.preventDefault();
		fetch(`https://api.openweathermap.org/data/2.5/forecast?q=${this.state.city},IN&APPID=1f703b39afd8359290cc27e826b45021`)
		.then(res => res.json())
		.then(data => this.setState({ weatherData: data }))
	}

	onChange = (event) =>  {
		this.setState({ city: event.target.value});
	}

	render() {
		let weather = _.get(this.state.weatherData, ['list']);
		return(
			<div>
				<form onSubmit={this.onSubmitCity}>
					<input type="text" name="city" value={this.state.city} onChange={this.onChange} />
					<button type="submit">Submit</button>
				</form>

				{this.state.city != null && 
					<div className="weather-data-container">
					    <h1 className="heading">5 day weather report for {this.state.city}</h1>
						<ul className="weather-item-list" style={{fontWeight: "bold"}}>
							<li key={"date"}>Date </li>
							<li key={"temp"}>Temp</li>
							<li key={"pre"}>Pressure</li>
							<li key={"hum"}>Humidity</li>
							<li key={"fore"}>Forecast</li>
							<li key={"wind"}>Wind Speed</li>
						</ul>
					{ weather && weather.map((entry,index) =>	
						<ul className="weather-item-list" key={index}>
							<li key={"date"+ index}> {entry.dt_txt} </li>
							<li key={"temp"+ index}>{(entry.main.temp - 273.15).toFixed(1)} &deg;C </li>
							<li key={"pre"+ index}> {parseInt(entry.main.pressure)} hpa</li>
							<li key={"hum"+ index}>{(entry.main.humidity)} %</li>
							<li key={"fore"+ index}> {entry.weather[0].description} </li>
							<li key={"wind"+ index}> {entry.wind.speed} m/s</li>
						</ul>
			 		)} 
			</div>}
		</div>
		)
	}
}
					

export default WeatherForecast;