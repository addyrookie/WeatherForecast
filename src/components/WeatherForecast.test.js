import  WeatherForecast from "./WeatherForecast";
import React from "react";
import { shallow } from "enzyme";

import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
configure({ adapter: new Adapter() });


describe('Testing presence of WeatherForecast component and its elements', () => {
	test('should render WeatherForecast component', () => {
		const wrapper =  shallow(<WeatherForecast />);
		expect(wrapper).toMatchSnapshot();
	})

	test('should contain a text box', () => {
		const wrapper =  shallow(<WeatherForecast />);
		expect(wrapper.find('input')).toMatchSnapshot();
	})

	test('should contain submit button', () => {
		const wrapper =  shallow(<WeatherForecast />);
		expect(wrapper.exists('button')).toBe(true)
	})
})


describe('Performing action on WeatherForecast', () => {
	test('weather forecast heading should display city name', () => {
		const wrapper =shallow(<WeatherForecast />); 
		wrapper.find('input').simulate('change', {target: {name: 'city', value: 'Kanpur' }})
		expect(wrapper.find('h1.heading').text()).toEqual('5 day weather report for Kanpur')
	})

	test('should update state value on change event', () => {
		const wrapper =shallow(<WeatherForecast />);
		const preventDefault = jest.fn();
		wrapper.find('input').simulate('change', {target: {name: 'city', value: 'Kanpur' }})
	 	expect(wrapper.state('city')).toEqual('Kanpur')
	})
})

