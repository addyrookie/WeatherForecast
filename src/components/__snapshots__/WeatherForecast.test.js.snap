// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing presence of WeatherForecast component and its elements should contain a text box 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <WeatherForecast />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        name="city"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
            <button
                        type="submit"
            >
                        Submit
            </button>
</form>,
          <div
            className="weather-data-container"
>
            <h1
                        className="heading"
            >
                        5 day weather report for 
                        
            </h1>
            <ul
                        className="weather-item-list"
                        style={
                                    Object {
                                                "fontWeight": "bold",
                                              }
                        }
            >
                        <li>
                                    Date 
                        </li>
                        <li>
                                    Temp
                        </li>
                        <li>
                                    Pressure
                        </li>
                        <li>
                                    Humidity
                        </li>
                        <li>
                                    Forecast
                        </li>
                        <li>
                                    Wind Speed
                        </li>
            </ul>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="city"
                onChange={[Function]}
                type="text"
                value=""
/>,
              <button
                type="submit"
>
                Submit
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "city",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="heading"
>
                5 day weather report for 
                
</h1>,
              <ul
                className="weather-item-list"
                style={
                                Object {
                                                "fontWeight": "bold",
                                              }
                }
>
                <li>
                                Date 
                </li>
                <li>
                                Temp
                </li>
                <li>
                                Pressure
                </li>
                <li>
                                Humidity
                </li>
                <li>
                                Forecast
                </li>
                <li>
                                Wind Speed
                </li>
</ul>,
              undefined,
            ],
            "className": "weather-data-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "5 day weather report for ",
                  "",
                ],
                "className": "heading",
              },
              "ref": null,
              "rendered": Array [
                "5 day weather report for ",
                "",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    Date 
</li>,
                  <li>
                    Temp
</li>,
                  <li>
                    Pressure
</li>,
                  <li>
                    Humidity
</li>,
                  <li>
                    Forecast
</li>,
                  <li>
                    Wind Speed
</li>,
                ],
                "className": "weather-item-list",
                "style": Object {
                  "fontWeight": "bold",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "date",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Date ",
                  },
                  "ref": null,
                  "rendered": "Date ",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "temp",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Temp",
                  },
                  "ref": null,
                  "rendered": "Temp",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "pre",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Pressure",
                  },
                  "ref": null,
                  "rendered": "Pressure",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "hum",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Humidity",
                  },
                  "ref": null,
                  "rendered": "Humidity",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "fore",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Forecast",
                  },
                  "ref": null,
                  "rendered": "Forecast",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "wind",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wind Speed",
                  },
                  "ref": null,
                  "rendered": "Wind Speed",
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <form
              onSubmit={[Function]}
>
              <input
                            name="city"
                            onChange={[Function]}
                            type="text"
                            value=""
              />
              <button
                            type="submit"
              >
                            Submit
              </button>
</form>,
            <div
              className="weather-data-container"
>
              <h1
                            className="heading"
              >
                            5 day weather report for 
                            
              </h1>
              <ul
                            className="weather-item-list"
                            style={
                                          Object {
                                                        "fontWeight": "bold",
                                                      }
                            }
              >
                            <li>
                                          Date 
                            </li>
                            <li>
                                          Temp
                            </li>
                            <li>
                                          Pressure
                            </li>
                            <li>
                                          Humidity
                            </li>
                            <li>
                                          Forecast
                            </li>
                            <li>
                                          Wind Speed
                            </li>
              </ul>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  name="city"
                  onChange={[Function]}
                  type="text"
                  value=""
/>,
                <button
                  type="submit"
>
                  Submit
</button>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "city",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="heading"
>
                  5 day weather report for 
                  
</h1>,
                <ul
                  className="weather-item-list"
                  style={
                                    Object {
                                                      "fontWeight": "bold",
                                                    }
                  }
>
                  <li>
                                    Date 
                  </li>
                  <li>
                                    Temp
                  </li>
                  <li>
                                    Pressure
                  </li>
                  <li>
                                    Humidity
                  </li>
                  <li>
                                    Forecast
                  </li>
                  <li>
                                    Wind Speed
                  </li>
</ul>,
                undefined,
              ],
              "className": "weather-data-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "5 day weather report for ",
                    "",
                  ],
                  "className": "heading",
                },
                "ref": null,
                "rendered": Array [
                  "5 day weather report for ",
                  "",
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li>
                      Date 
</li>,
                    <li>
                      Temp
</li>,
                    <li>
                      Pressure
</li>,
                    <li>
                      Humidity
</li>,
                    <li>
                      Forecast
</li>,
                    <li>
                      Wind Speed
</li>,
                  ],
                  "className": "weather-item-list",
                  "style": Object {
                    "fontWeight": "bold",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "date",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Date ",
                    },
                    "ref": null,
                    "rendered": "Date ",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": "temp",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Temp",
                    },
                    "ref": null,
                    "rendered": "Temp",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": "pre",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Pressure",
                    },
                    "ref": null,
                    "rendered": "Pressure",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": "hum",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Humidity",
                    },
                    "ref": null,
                    "rendered": "Humidity",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": "fore",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Forecast",
                    },
                    "ref": null,
                    "rendered": "Forecast",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": "wind",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Wind Speed",
                    },
                    "ref": null,
                    "rendered": "Wind Speed",
                    "type": "li",
                  },
                ],
                "type": "ul",
              },
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
      "attachTo": undefined,
      "hydrateIn": undefined,
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "name": "city",
      "onChange": [Function],
      "type": "text",
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "name": "city",
        "onChange": [Function],
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Testing presence of WeatherForecast component and its elements should render WeatherForecast component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherForecast />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    name="city"
                    onChange={[Function]}
                    type="text"
                    value=""
          />
          <button
                    type="submit"
          >
                    Submit
          </button>
</form>,
        <div
          className="weather-data-container"
>
          <h1
                    className="heading"
          >
                    5 day weather report for 
                    
          </h1>
          <ul
                    className="weather-item-list"
                    style={
                              Object {
                                        "fontWeight": "bold",
                                      }
                    }
          >
                    <li>
                              Date 
                    </li>
                    <li>
                              Temp
                    </li>
                    <li>
                              Pressure
                    </li>
                    <li>
                              Humidity
                    </li>
                    <li>
                              Forecast
                    </li>
                    <li>
                              Wind Speed
                    </li>
          </ul>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="city"
              onChange={[Function]}
              type="text"
              value=""
/>,
            <button
              type="submit"
>
              Submit
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "city",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="heading"
>
              5 day weather report for 
              
</h1>,
            <ul
              className="weather-item-list"
              style={
                            Object {
                                          "fontWeight": "bold",
                                        }
              }
>
              <li>
                            Date 
              </li>
              <li>
                            Temp
              </li>
              <li>
                            Pressure
              </li>
              <li>
                            Humidity
              </li>
              <li>
                            Forecast
              </li>
              <li>
                            Wind Speed
              </li>
</ul>,
            undefined,
          ],
          "className": "weather-data-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "5 day weather report for ",
                "",
              ],
              "className": "heading",
            },
            "ref": null,
            "rendered": Array [
              "5 day weather report for ",
              "",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  Date 
</li>,
                <li>
                  Temp
</li>,
                <li>
                  Pressure
</li>,
                <li>
                  Humidity
</li>,
                <li>
                  Forecast
</li>,
                <li>
                  Wind Speed
</li>,
              ],
              "className": "weather-item-list",
              "style": Object {
                "fontWeight": "bold",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "date",
                "nodeType": "host",
                "props": Object {
                  "children": "Date ",
                },
                "ref": null,
                "rendered": "Date ",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "temp",
                "nodeType": "host",
                "props": Object {
                  "children": "Temp",
                },
                "ref": null,
                "rendered": "Temp",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "pre",
                "nodeType": "host",
                "props": Object {
                  "children": "Pressure",
                },
                "ref": null,
                "rendered": "Pressure",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "hum",
                "nodeType": "host",
                "props": Object {
                  "children": "Humidity",
                },
                "ref": null,
                "rendered": "Humidity",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "fore",
                "nodeType": "host",
                "props": Object {
                  "children": "Forecast",
                },
                "ref": null,
                "rendered": "Forecast",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "wind",
                "nodeType": "host",
                "props": Object {
                  "children": "Wind Speed",
                },
                "ref": null,
                "rendered": "Wind Speed",
                "type": "li",
              },
            ],
            "type": "ul",
          },
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        name="city"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
            <button
                        type="submit"
            >
                        Submit
            </button>
</form>,
          <div
            className="weather-data-container"
>
            <h1
                        className="heading"
            >
                        5 day weather report for 
                        
            </h1>
            <ul
                        className="weather-item-list"
                        style={
                                    Object {
                                                "fontWeight": "bold",
                                              }
                        }
            >
                        <li>
                                    Date 
                        </li>
                        <li>
                                    Temp
                        </li>
                        <li>
                                    Pressure
                        </li>
                        <li>
                                    Humidity
                        </li>
                        <li>
                                    Forecast
                        </li>
                        <li>
                                    Wind Speed
                        </li>
            </ul>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="city"
                onChange={[Function]}
                type="text"
                value=""
/>,
              <button
                type="submit"
>
                Submit
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "city",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="heading"
>
                5 day weather report for 
                
</h1>,
              <ul
                className="weather-item-list"
                style={
                                Object {
                                                "fontWeight": "bold",
                                              }
                }
>
                <li>
                                Date 
                </li>
                <li>
                                Temp
                </li>
                <li>
                                Pressure
                </li>
                <li>
                                Humidity
                </li>
                <li>
                                Forecast
                </li>
                <li>
                                Wind Speed
                </li>
</ul>,
              undefined,
            ],
            "className": "weather-data-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "5 day weather report for ",
                  "",
                ],
                "className": "heading",
              },
              "ref": null,
              "rendered": Array [
                "5 day weather report for ",
                "",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    Date 
</li>,
                  <li>
                    Temp
</li>,
                  <li>
                    Pressure
</li>,
                  <li>
                    Humidity
</li>,
                  <li>
                    Forecast
</li>,
                  <li>
                    Wind Speed
</li>,
                ],
                "className": "weather-item-list",
                "style": Object {
                  "fontWeight": "bold",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "date",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Date ",
                  },
                  "ref": null,
                  "rendered": "Date ",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "temp",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Temp",
                  },
                  "ref": null,
                  "rendered": "Temp",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "pre",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Pressure",
                  },
                  "ref": null,
                  "rendered": "Pressure",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "hum",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Humidity",
                  },
                  "ref": null,
                  "rendered": "Humidity",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "fore",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Forecast",
                  },
                  "ref": null,
                  "rendered": "Forecast",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "wind",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wind Speed",
                  },
                  "ref": null,
                  "rendered": "Wind Speed",
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
